### ISSUE TITLE
pylint fails to check decorator's arguments

### ISSUE THREAD

**Issue by zus0** [PyCQA/pylint#3772]
### Steps to reproduce
```python   
def my_decorator(func):
    def wrapper(*args, **kwargs):
        print("Something is happening before the function is called.")
        func(['some', 'list'], *args, **kwargs)
        print("Something is happening after the function is called.")
    return wrapper

@my_decorator
def say_whee(arg, arg2):
    print(arg)
    print(arg2, "Whee!")

say_whee('Hello')
```

### Current behavior
`E1120: No value for argument 'arg2' in function call (no-value-for-parameter)`

### Expected behavior
No errors

### pylint --version output
```
pylint 2.5.3
astroid 2.4.2
Python 3.8.2 (default, Jul 16 2020, 14:00:26) 
[GCC 9.3.0]
```

**Comment by AWhetter** [comment]
There's a `signature-mutators` option that will disable `too-many-function-args`, `unexpected-keyword-arg`, and `no-value-for-parameter` for functions decorated with any of the specified decorators. If you set that option correctly, then the warning should go away.
